lib/timers.js

```diff
@@ -87,13 +87,20 @@ function listOnTimeout() {
 
   var first;
   while (first = L.peek(list)) {
-    // If the previous iteration caused a timer to be added,
-    // update the value of "now" so that timing computations are
-    // done correctly. See test/simple/test-timers-blocking-callback.js
-    // for more information.
+    // Whenever a timer creates a new timer in the current list being processed
```

Comment:
`you can end up processing the new timer immediately instead of waiting the scheduled time.`
is a bit too vague. I would suggest rephrasing it as following:

```
This handles the case of a timer that was created within a timers callback with the same timeout value.
For instance, when processing the timer that would call `bar` in such code :

setTimeout(function foo() { setTimeout(function bar() {}, 0) }, 0);

or 

setTimeout(function foo() { setTimeout(function bar() {}, 5000) }, 5000);

We want to make sure that newly added timer fires in the next turn of the event loop
at the earliest. So even if it's already expired now, reschedule it to fire later.

At that point, it's not necessary to process any other timer in that list,
because any remaining timer has been added within a callback of a timer
that has already been processed, and thus needs to be processed at the
earliest not in the current tick, but when the rescheduled timer will expire.
```

Basically we want to be as explicit and precise as possible, so that when we come back to this piece of code
in 6 months, we understand why we made this change.